model_templates/training/python3_sparse/custom.py

```diff
@@ -0,0 +1,148 @@
+import pickle
+from typing import List, Optional
+
+import numpy as np
+import pandas as pd
+from scipy.sparse import csr_matrix
+from sklearn.linear_model import Ridge
+
+
+def fit(
+    X: csr_matrix,
+    y: pd.Series,
+    output_dir: str,
+    class_order: Optional[List[str]] = None,
+    row_weights: Optional[np.ndarray] = None,
+    **kwargs,
+):
+    """
+    This hook must be implemented with your fitting code, for running drum in the fit mode.
+
+    This hook MUST ALWAYS be implemented for custom training models.
+    For inference models, this hook can stick around unimplemented, and won’t be triggered.
+
+    Parameters
+    ----------
+    X: pd.DataFrame - training data to perform fit on
```

Comment:
nit- a couple of comments/docstrings here could use updating (I think just X's expect type and the comment below about "feel free to delete...")

Reply:
I don't understand what you mean by this.

Reply:
the param for X in the docstring says we're expecting a pd.DataFrame - not sure if that's true since the fit call says it's a csr_matrix? and the line above estimator=Ridge() that says "feel free to delete whichever of these you aren't using" when there's only mod estimator path...I'm pretty sure our templates are full of little stuff like that

Reply:
good comments.